### DISCUSSION TITLE
Add button to reset column visibility in tables

### DISCUSSION THREAD

**Discussion by buddhaCode** [filamentphp/filament#11736] [Ideas]
Hi there,
I have toggle-able columns in my table. I would like to reset the visibility of these columns to the default state. Like resetting the filters in the filter panel.

This was [already discusses in Discord](https://www.answeroverflow.com/m/1139283948594282496) some time ago.

I could deliver the PR. @danharrin what do you think? Could this be a suitable feature?

Current state:

<img width="365" alt="Bildschirmfoto 2024-03-07 um 09 59 38" src="https://github.com/filamentphp/filament/assets/11678100/320b8948-738b-486b-9462-da986f2529dc">

Proposed state:

<img width="367" alt="Bildschirmfoto 2024-03-07 um 10 01 01" src="https://github.com/filamentphp/filament/assets/11678100/50e7a505-4a05-4a19-8d90-1f9e0eafa223">

Best, Arne

**Comment by danharrin** [comment]
Sounds good to me

**Comment by niladam** [comment]
Hello @buddhaCode -- i was wondering if you had a chance to raise a PR for this ? Are you still looking into PRing it ?

**Reply by buddhaCode** [reply]
Dear @niladam,
I have tried to implement this function. But I found the structure of the panel component quite complex. In the end, something else got in the way. So I didn't pursue the whole thing any further.
